### ISSUE TITLE
Lighting patch overwrites subclassed `TensorImage` type to `TensorImage`

### ISSUE THREAD

**Issue by kai-tub** [fastai/fastai#3246]
First asked on the forum [here](https://forums.fast.ai/t/bug-lightning-patch-overwrites-type/86352).

Please confirm you have the latest versions of fastai, fastcore, and nbdev prior to reporting a bug (delete one): YES

**Describe the bug**
The lightning transform is patched to all `TensorImage` types but converts the input to a `TensorImage` and breaks sub-classed `TensorImage` code. 
See:

https://github.com/fastai/fastai/blob/master/fastai/vision/augment.py#L668

**To Reproduce**
To reproduce and see my proposed fix take a look at the following Colab notebook:

https://colab.research.google.com/drive/1Mexm84QcKNEag1CBmrI70SJ95pyz6YTU?usp=sharing

TL;DR:
Change

```python
@patch
def lighting(x: TensorImage, func): return TensorImage(torch.sigmoid(func(logit(x))))
```

to

```python
@patch
def lighting(x: TensorImage, func): return torch.sigmoid(func(logit(x))
```

But I may be missing why the conversion to TensorImage is necessary.
For me at least, it is unexpected behavior and breaks my subclass.

Would be more than happy to create a tiny PR and hear some comments :slight_smile:
